druid/src/text/editable_text.rs

```diff
@@ -156,6 +162,32 @@ impl EditableText for String {
     fn from_str(s: &str) -> Self {
         s.to_string()
     }
+
+    fn line_beginning(&self, from: usize) -> Option<usize> {
+        let mut offset = from;
+
+        for grapheme in self.get(0..from)?.graphemes(true).rev() {
+            if grapheme.chars().next()? == '\n' {
+                return Some(offset);
+            }
+            offset -= grapheme.len();
+        }
```

Comment:
We don't need to be fancy with graphemes here; we're only looking for the 0x80 (`\n`) byte, and we can must iterate the bytes directly, which is much cheaper.

Additionally I think this doesn't need to return an `Option`; it can return `0` if there's no preceding newline.

Reply:
`\n` is 0x0a